### PR TITLE
Async quest patch

### PR DESCRIPTION
**Pull request by PhoenixICE**
Automatically determines which Map's async time to use if a quest targets a map
Uses that AsyncTime to tick for all aspects of that quest
If no map is targeted then use the world time as a fall back
Does not change Quest generation, it is still based on world time

### CODE REVIEW

**Comment by PhoenixICE** [comment]
New commit should resolve all the issues you had, let me know if there are any other issues.

**Comment by notfood** [comment]
Shouldn't we merging this to unstable 1.1?

File: Source/Client/Comp/MultiplayerAsyncQuest.cs [hunk 1]
```diff
@@ -135,24 +133,37 @@ public static class MultiplayerAsyncQuest
...
+        /// Tries to remove Map from cache, then moves all Quests cached to that map to WorldQuestsCache
+        /// </summary>
+        /// <param name="mapAsyncTimeComp">Map to remove</param>
+        /// <returns>If map is found in cache</returns>
+        public static bool TryRemoveCachedMap(MapAsyncTimeComp mapAsyncTimeComp)
         {
-            if (!MultiplayerWorldComp.asyncTime || mapAsyncTimeComp == null) return null;
-            mapQuestsCache.TryGetValue(mapAsyncTimeComp, out var quests);
-            return quests;
+            if (mapQuestsCache.TryGetValue(mapAsyncTimeComp, out var quests))
```
Comment (Nebual) [inline Source/Client/Comp/MultiplayerAsyncQuest.cs:143]: It looks like this method no longer actually removes from mapQuestsCache?

Comment (PhoenixICE) [reply Source/Client/Comp/MultiplayerAsyncQuest.cs:143]: Ahh damnit must've lost it in all the refactoring, thanks for spotting the error.

**Comment by notfood** [comment]
How stable is this? Ready for merge?

**Comment by PhoenixICE** [comment]
Afaik its already in Unstable 1.1 branch

But from a bug perspective, there is still one outstanding issue.

The cache needs be updated when a Quest object is deleted, it currently just holds the object forever in memory, its not a huge issue since if you restart/reload it will disappear - but as Nebula pointed out it could cause problems if we ever loop through cache quests and attempt to do something with them.

**Comment by notfood** [comment]
Unstable 1.1 is for experiments, sometimes they don't get included.

**Comment by PhoenixICE** [comment]
I haven't had any bug reports from users so either there are no bugs, no one uses beta16 or no one plays async time.

The bug fixes have been from my own games so far.

**Pull request merged**